### PR TITLE
Add `ReduceMotionConfig` docs

### PR DESCRIPTION
**Pull request by piaskowyk**
## Summary

Documentation for component `ReduceMotionConfig` added in https://github.com/software-mansion/react-native-reanimated/pull/6164

image

https://github.com/software-mansion/react-native-reanimated/pull/6164 - needs to be merged before

https://github.com/user-attachments/assets/bc6e467b-5ab3-4e77-94b6-7d792541643f

### CODE REVIEW

File: packages/docs-reanimated/src/examples/ReduceMotionConfig.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  useEffect(() => {
+    sv.value = withRepeat(withTiming(360, { duration: 2000 }), -1, true);
+  });
+
+  return (
+    <View style={styles.container}>
+      {/* I'll uncomment if after merge of ReduceMotionConfig to main and new nightly release, because now component is unavailable */}
+      {/* <ReduceMotionConfig mode={ReduceMotion.Never} /> */}
+      <Animated.View style={[styles.box, animatedStyle]} />
+    </View>
```
Comment (szydlovsky) [inline packages/docs-reanimated/src/examples/ReduceMotionConfig.tsx:27]: If possible, I'd add some simple reduce motion mode switch to the interactive example. Some less Reanimated-experienced docs users won't be able to tell if the animation worked before applying the `.Never` mode or not

**Review by szydlovsky — APPROVED** [review]
Short and nicely informational! I have one potential improvement idea but it is not a must-have

File: packages/docs-reanimated/src/examples/ReduceMotionConfig.tsx (hunk 1 above)
Comment (piaskowyk) [reply packages/docs-reanimated/src/examples/ReduceMotionConfig.tsx:27]: make sense, I'll do it 🫡

File: packages/docs-reanimated/docs/components/ReducedMotionConfig.mdx [hunk 2]
```diff
@@ -0,0 +1,66 @@
+---
+id: ReducedMotionConfig
+title: ReducedMotionConfig
+sidebar_label: ReducedMotionConfig
+sidebar_position: 1
+---
+
+The `ReducedMotionConfig` component determines the default animation behavior in response to the device's reduced motion accessibility setting. It affects your application globally. The default behavior disables all animation if reduced motion is enabled on a device. You can utilize this component to override that behavior. If you wish to fine-tune it for your specific case, learn more about [Accessibility](../guides/accessibility) and [`useReducedMotion`](../device/useReducedMotion) in Reanimated.
```
Comment (kacperkapusciak) [inline packages/docs-reanimated/docs/components/ReducedMotionConfig.mdx:8]:
```suggestion
`ReducedMotionConfig` component let's you change behavior in response to the device's reduced motion accessibility setting. By default it disables all animation when the reduced motion is enabled on a device. You can adjust it for your specific use case. You can learn more about [Accessibility](/docs/guides/accessibility) and [`useReducedMotion`](/docs/device/useReducedMotion) in Reanimated.
```

File: packages/docs-reanimated/src/examples/ReducedMotionConfig.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  const [isReduceMotionDisabled, setIsReduceMotionDisabled] = useState(false);
+  const sv = useSharedValue<number>(0);
+  const animatedStyle = useAnimatedStyle(() => ({
+    transform: [{ rotate: `${sv.value}deg` }],
+  }));
+
+  useEffect(() => {
+    sv.value = 0;
+    sv.value = withRepeat(withTiming(360, { duration: 2000 }), -1, true);
+  });
```
Comment (kacperkapusciak) [inline packages/docs-reanimated/src/examples/ReducedMotionConfig.tsx:24]: i haven't run this but shouldn't there be a dependency array here?

**Comment by mzaien** [comment]
Hi @piaskowyk I wanted to know, can we use it per component? or is it a global config

**Pull request merged**